app/src/main/java/com/keylesspalace/tusky/viewmodel/EditProfileViewModel.kt

```diff
@@ -92,62 +79,20 @@ class EditProfileViewModel @Inject constructor(
                         profileData.postValue(Error())
                     }
                 )
-                .addTo(disposeables)
+                .addTo(disposables)
         }
     }
 
-    fun newAvatar(uri: Uri, context: Context) {
-        val cacheFile = getCacheFileForName(context, AVATAR_FILE_NAME)
-
-        resizeImage(uri, context, AVATAR_SIZE, AVATAR_SIZE, cacheFile, avatarData)
-    }
+    fun getAvatarUri(context: Context) = getCacheFileForName(context, AVATAR_FILE_NAME).toUri()
```

Comment:
We did this before but it's a bit eh that we pass context here, especially because any context will do here, including global

Reply:
Yes I know, but injecting context into viewmodel gives a warning because of leaked context 🙄

Reply:
AndroidViewModel but it has constructor restrictions
https://developer.android.com/reference/androidx/lifecycle/AndroidViewModel

Reply:
well, why not just inject an Application instead of a context into the regular Viewmodel

Reply:
looks good

Reply:
Yup, I think that's better than risking to leak Activity